compiler/rustc_hir/src/def.rs

```diff
@@ -341,9 +341,7 @@ impl<T> PerNS<Option<T>> {
 
     /// Returns an iterator over the items which are `Some`.
     pub fn present_items(self) -> impl Iterator<Item = T> {
-        use std::iter::once;
-
-        once(self.type_ns).chain(once(self.value_ns)).chain(once(self.macro_ns)).filter_map(|it| it)
+        IntoIter::new([self.type_ns, self.value_ns, self.macro_ns]).filter_map(|it| it)
```

Comment:
Could have been `.flatten()`?

Reply:
Frustratingly, `flatten` gives worse `size_hint`s than `filter_map` today (https://github.com/rust-lang/rust/pull/48544) and has more complicated code (needs to keep the iterator in its state after giving out an item, since unlike `Option` it doesn't know if that's the only item), so that's potentially a pessimization.